### PR TITLE
Fix FilteringProvider to accept Spec instead of Predicate

### PR DESCRIPTION
**Pull request by ov7a**
### Context
It was missed during the [original PR](https://github.com/gradle/gradle/pull/26067) for
* https://github.com/gradle/gradle/issues/19981

Reported by @lacasseio from the Nokee project.

### Reviewing cheatsheet

Before merging the PR, comments starting with 
- ❌ ❓**must** be fixed
- 🤔 💅 **should** be fixed
- 💭 **may** be fixed
- 🎉 celebrate happy things

### CODE REVIEW

**Review by ljacomet — CHANGES_REQUESTED** [review]
Code changes LGTM

But since we are changing an API, even incubating, I think we should have a mention of the impact in the upgrade guide under potential breaking changes.

**Comment by ov7a** [comment]
@bot-gradle test this

**Review by ljacomet — APPROVED** [review]

File: subprojects/docs/src/docs/userguide/releases/upgrading/upgrading_version_8.adoc [hunk 1]
```diff
@@ -47,6 +47,12 @@ The `groovy-base` plugin is now responsible for configuring source and target co
 If you are using this task *without applying `grooy-base`*, you will have to manually set compatibility versions on these tasks.
 In general, the `groovy-base` plugin should be applied whenever working with Groovy language tasks.
 
+==== Provider.filter
+
+The type of the argument passed to `Provider.filter` is changed from `Predicate` to `Spec` to have a more consistent API.
+This change should not affect anyone using `Provider.filter` with a lambda expression.
```
Comment (lkasso) [inline subprojects/docs/src/docs/userguide/releases/upgrading/upgrading_version_8.adoc:53]:
```suggestion
This change does not affect anyone using `Provider.filter` with a lambda expression.
```

File: subprojects/docs/src/docs/userguide/releases/upgrading/upgrading_version_8.adoc [hunk 2]
```diff
@@ -47,6 +47,12 @@ The `groovy-base` plugin is now responsible for configuring source and target co
 If you are using this task *without applying `grooy-base`*, you will have to manually set compatibility versions on these tasks.
 In general, the `groovy-base` plugin should be applied whenever working with Groovy language tasks.
 
+==== Provider.filter
+
+The type of the argument passed to `Provider.filter` is changed from `Predicate` to `Spec` to have a more consistent API.
```
Comment (lkasso) [inline subprojects/docs/src/docs/userguide/releases/upgrading/upgrading_version_8.adoc:52]:
```suggestion
The type of the argument passed to `Provider.filter` is changed from `Predicate` to `Spec` for a more consistent API.
```

File: subprojects/docs/src/docs/userguide/releases/upgrading/upgrading_version_8.adoc [hunk 3]
```diff
@@ -47,6 +47,12 @@ The `groovy-base` plugin is now responsible for configuring source and target co
...
+This change should not affect anyone using `Provider.filter` with a lambda expression.
+However, this might have consequences for plugin authors.
```
Comment (lkasso) [inline subprojects/docs/src/docs/userguide/releases/upgrading/upgrading_version_8.adoc:54]: Can we detail the consequences? What should they do?

**Review by lkasso — APPROVED** [review]
Docs are approved with a few notes.

**Comment by ov7a** [comment]
@bot-gradle test and merge

**Pull request merged**